UndertaleModLib/Decompiler/Instructions/Decompiler.AssignmentStatement.cs

```diff
@@ -117,7 +117,16 @@ bool checkEqual(ExpressionVar a, ExpressionVar b)
                     }
                 }
             }
-            return String.Format("{0}{1}{2} {3}", varPrefix, varName, context.DecompilingStruct ? ":" : " =", Value.ToString(context));
+
+            // Since parenthesized is now the default output of ExpressionTwo (mainly math operations)
+            // It is specifically safe in variable assignment to not parenthesize this (e.g. "foo = a + b")
+            string stringOfValue;
+            if (Value is ExpressionTwo expressionTwoVal)
```

Comment:
So what if you have `foo = (a + b) * 4`? Would that be properly shown like that, or shown as `foo = a + b * 4` and then cause further errors down the line?

Reply:
That would evaluate as an ExpressionTwo where its first argument is an ExpressionTwo of lower precedence. This only removes the outermost parentheses from `foo = ((a + b) * 4)`.

Reply:
ah right.